internal/plugin/connectors/http/basicauth/plugin.go

```diff
@@ -16,16 +18,22 @@ func PluginInfo() map[string]string {
 	}
 }
 
-// NewConnector returns an http.Connector that decorates each incoming http
-// request with a basic auth header.
-func NewConnector(conRes connector.Resources) http.Connector {
-	return &Connector{
-		logger:   conRes.Logger(),
-	}
-}
-
 // GetHTTPPlugin is required as part of the Secretless plugin spec for HTTP
 // connector plugins. It returns the HTTP plugin.
 func GetHTTPPlugin() http.Plugin {
-	return http.ConnectorConstructor(NewConnector)
+	newConnector, err := generichttp.NewNewConnector(
+		&generichttp.ConfigYAML{
+			CredentialPatterns: map[string]string{
+				"username": "[^:]+",
+			},
+			Headers: map[string]string{
+				"Authorization": " Basic {{ printf \"%s:%s\" .username .password | base64 }}",
```

Comment:
The prefix space in `" Basic ...` is either an error or we should figure out why it's needed - seems odd that it would be needed.

Reply:
Total newb questions but is this evaluated here or is something else evaluating that dynamic string somewhere?

Reply:
It's a go text template and template itself is created in the `NewNewConnector` ctor.  That template remains in memory and is rendered with the credentials at runtime.

Reply:
k - I think we still need to remove the prefixed whitespace though, right?

Reply:
Yeah I'll try removing it.  It should be fine I think.  I only put it there because its generally there if you look at the raw headers.